lib/tapioca/dsl/compiler.rb

```diff
@@ -98,8 +98,10 @@ def parameters_types_from_signature(method_def, signature)
         # keyword parameters types
         signature.kwarg_types.each { |_, kwarg_type| params << kwarg_type.to_s }
 
-        # rest parameter type
-        params << signature.rest_type.to_s if signature.has_rest
+        # rest and keyrest parameter types
+        ["", "key"].each do |prefix|
+          params << signature.public_send("#{prefix}rest_type").to_s if signature.public_send("has_#{prefix}rest")
+        end
```

Comment:
I'm not convince with the use of `public_send` here. Wouldn't calling the real method directly make it easier to read and typecheck?

Something like this:

```rb
        # rest parameter type
        params << signature.rest_type.to_s if signature.has_rest

        # keyrest parameter type
        params << signature.keyrest_type.to_s if signature.has_keyrest
```

Reply:
Agree, removed the `public_send`